src/main.py

```
@@ -1,17 +1,49 @@
 
+
+def wait_for(host, port, timeout=20):
+    start_time = time.time()
+    while True:
+        try:
+            with socket.create_connection((host, port), timeout=2):
+                print(f"Connected to {host}:{port}!")
+                return
+        except OSError:
+            if time.time() - start_time > timeout:
+                raise TimeoutError(f"Timed out waiting for {host}:{port}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)